src/peer_manager.rs

```diff
@@ -536,28 +554,39 @@ impl PeerManager {
         }
     }
 
-    /// Returns a tuple containing the verified candidate's `PublicId`, its client `Authority` and
-    /// the `PublicId`s of all routing table entries.
-    pub fn verified_candidate_info
-        (&self)
-         -> Result<(PublicId, Authority<XorName>, SectionMap), RoutingError> {
-        if let Some((name, candidate)) =
+    /// Returns a (`MessageContent::CandidateApproval`, new name) tuple completed using the verified
+    /// candidate's details.
+    pub fn verified_candidate_info(&self) -> Result<(MessageContent, XorName), RoutingError> {
+        if let Some((old_pub_id, candidate)) =
             self.candidates
                 .iter()
                 .find(|&(_, cand)| cand.passed_our_challenge && !cand.is_approved()) {
-            return if let Some(peer) = self.peer_map.get_by_name(name) {
-                       Ok((*peer.pub_id(), candidate.client_auth, self.pub_ids_by_section()))
+            return if let (Some(new_pub_id), Some(new_client_auth)) =
+                (candidate.new_pub_id.as_ref(), candidate.new_client_auth.as_ref()) {
+                       self.peer_map
+                           .get_by_name(new_pub_id.name())
+                           .map_or(Err(RoutingError::UnknownCandidate), |_peer| {
+                    Ok((MessageContent::CandidateApproval {
+                            old_public_id: *old_pub_id,
+                            new_public_id: *new_pub_id,
+                            new_client_auth: *new_client_auth,
+                            sections: self.pub_ids_by_section(),
+                        },
+                        *new_pub_id.name()))
+                })
```

Comment:
can we not to put this competed stuff inside a return clause?
also seems the result from rustfmt is not good.  :(